api/filters/sharing_instance_filter.go

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/Peripli/service-manager/pkg/instance_sharing"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+	"net/http"
+)
+
+const SharingInstanceFilterName = "SharingInstanceFilter"
+
+// SharingInstanceFilter validate the un/share request on an existing service instance
+type sharingInstanceFilter struct {
+	storageRepository storage.Repository
+}
+
+func NewSharingInstanceFilter(storageRepository storage.Repository) *sharingInstanceFilter {
+	return &sharingInstanceFilter{
+		storageRepository: storageRepository,
+	}
+}
+
+func (*sharingInstanceFilter) Name() string {
+	return SharingInstanceFilterName
+}
+
+func (sf *sharingInstanceFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
+	return sf.handleServiceUpdate(req, next)
+}
+
+func (*sharingInstanceFilter) FilterMatchers() []web.FilterMatcher {
+	return []web.FilterMatcher{
+		{
+			Matchers: []web.Matcher{
+				web.Path(web.ServiceInstancesURL + "/**"),
+				web.Methods(http.MethodPatch),
+			},
+		},
+	}
+}
+
+func (sf *sharingInstanceFilter) validateOnlySharedPropertyIsChanged(persistedInstance *types.ServiceInstance, reqInstanceBytes *[]byte) error {
+	var updatedInstance types.ServiceInstance
+	persistedInstanceBytes, err := json.Marshal(&persistedInstance)
+	if err != nil {
+		return err
+	}
+	if err := util.BytesToObject(persistedInstanceBytes, &updatedInstance); err != nil {
+		return err
+	}
+	if err := util.BytesToObject(*reqInstanceBytes, &updatedInstance); err != nil {
+		return err
+	}
+
+	//in order to ignore shared property when validating the request we set it to be equals
+	//TODO: find out why the context is not the same (the persisted instance has instance_name property and updatedInstance does not have it)
+	updatedInstance.Shared = persistedInstance.Shared
+	updatedInstance.Context = persistedInstance.Context
+
+	if !persistedInstance.Equals(&updatedInstance) {
```

Comment:
does this comparison catch changes in parameters? i guess not as they are not part of the storage entity

Reply:
we have changed the implementation to confirm that just a single "shared" instance property exists as part of the request body. incase it does not exist the filter is skipped, incase other properties exist, we will fail the request.